core/divelist.c

```diff
@@ -495,6 +499,12 @@ int init_decompression(struct deco_state *ds, const struct dive *dive, bool in_p
 		printf("Check if dive #%d %d has to be considered as prev dive: ", i, get_dive(i)->number);
 #endif
 		struct dive *pdive = get_dive(i);
+		if (!pdive || pdive->when >= dive->when || dive_endtime(pdive) + 48 * 60 * 60 < last_starttime) {
+#if DECO_CALC_DEBUG & 2
+			printf("No\n");
+#endif
+			break;
+		}
 		/* we don't want to mix dives from different trips as we keep looking
 		 * for how far back we need to go */
 		if (dive->divetrip && pdive->divetrip != dive->divetrip) {
```

Comment:
If the divetrip check is done first then we end up going through the whole dive list if all the dives in the trip are within 48 hours.